weatherman.py

```diff
@@ -0,0 +1,201 @@
+    print(f"Most humidity {most_humidity}% on  {most_humidity_day} ")
+
+
+def print_monthly_weather_record(mean_iteration, mean_humidity, average_lowest_temperature, average_highest_temperature):
+    if mean_iteration > 0:
+        average_mean_iteration = mean_humidity / mean_iteration
+        average_highest_temperature /= mean_iteration
+        average_lowest_temperature /= mean_iteration
+
+        print(f"Average Lowest Temperature: {average_lowest_temperature:.3f}")
+        print(f"Average Highest Temperature: {average_highest_temperature:.3f}")
+        print(f"Average Mean Humidity: {average_mean_iteration:.3f}%")
+    else:
+        print("No valid data found for the given month.")
+
+
+def print_weather_bar_chart(weather_records):
+    for weather_record in weather_records:
+        date = weather_record['PKT']
+
+        try:
+            maximum_temperature = int(weather_record['Max TemperatureC'])
+            minimum_temperature = int(weather_record['Min TemperatureC'])
+            reset_code = Style.RESET_ALL
+
+            color_code = Fore.RED
+            bar = "+" * maximum_temperature
+            print(f"{color_code}{date} {bar:<1} {maximum_temperature}C{reset_code}")
+
+            color_code = Fore.BLUE
+            bar = "+" * minimum_temperature
+            print(f"{color_code}{date} {bar:<1} {minimum_temperature}C{reset_code}")
+
+        except ValueError:
+            continue
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="Process weatherman data files.")
+    parser.add_argument(
+        "-e", "--weather_year",
+        help="For a given year, it will display the highest temperature and day,"
+             " lowest temperature and day, most humid day and humidity."
+    )
+    parser.add_argument(
+        "-a", "--weather_year_month",
+        help="For a given month, it will display the average highest temperature,"
+             " average lowest temperature, average mean humidity."
+    )
+    parser.add_argument(
+        "-c", "--weather_bar_chart",
+        help="For a given month, it will draw two horizontal bar charts on the console"
+             " for the highest and lowest temperature on each day. Highest in"
+             " red and lowest in blue."
+    )
+    return parser.parse_args()
+
+
+def main():
+    parsed_arguments = parse_arguments()
+    weather_year = None
+    weather_month = None
+    try:
+        if parsed_arguments.weather_year:
+            weather_year = parsed_arguments.weather_year
+            weather_records = read_weather_files(weather_year, weather_month)
+            statistics_year = calculate_yearly_weather_records(weather_records)
+            print_yearly_weather_record(**statistics_year)
+
+        if parsed_arguments.weather_year_month:
+            weather_year, weather_month_input = parsed_arguments.weather_year_month.split('/')
+            try:
+                weather_month = convert_month(weather_month_input)
```

Comment:
Why do we duplicate this method in every if block? If it's utilized in each block, it would be better to invoke it at the beginning of the if blocks.

Reply:
talked to you in person about this